marvis/node/docker.py

```diff
@@ -223,12 +223,17 @@ def start_docker_container(self, log_directory, hosts=None):
 
     def stop_docker_container(self):
         """Stop the container."""
-        if self.container is not None:
-            logger.info('Stopping docker container: %s', self.container.name)
-            self.container.stop(timeout=1)
-            self.container = None
-            self.container_pid = None
-            self.command_executor = None
+        if self.container is None:
+            return
+        logger.info('Stopping docker container: %s', self.container.name)
+        if self.container.status != 'running':
```

Comment:
Hm what if the container is `restarting` or `paused`?

Reply:
fair… I'll switch to `try … except` (which I did not do in the first place, since I find guard clauses result in better readability)

Reply:
On a second thought, errors should not pass silently and there should be a way to prevent marvis from automatically stopping containers.

Take the SUMO example, where the application container exits after it has done its job. This seems to be on purpose and it should not be attempted to stop the container.

Reply:
Where do we go from here? The changes above at least prevent an error I encountered.